findimports.py

```diff
@@ -592,18 +624,23 @@ def parseFile(self, filename, ignore_stdlib_modules):
             module.imported_names, module.unused_names = (
                 find_imports_and_track_names(filename,
                                              self.warn_about_duplicates,
-                                             self.verbose)
+                                             self.verbose,
+                                             self.max_depth)
             )
         else:
-            module.imported_names = find_imports(filename)
+            module.imported_names = find_imports(filename, self.max_depth)
             module.unused_names = None
         dir = os.path.dirname(filename)
+
+        if ignore_stdlib_modules:
+            module.imported_names = [
+                info for info in module.imported_names
+                if info.name.split('.')[0] not in STDLIB_MODNAMES_SET
+            ]
         module.imports = {
             self.findModuleOfName(imp.name, imp.level, filename, dir)
             for imp in module.imported_names}
         if ignore_stdlib_modules:
-            module.imported_names = [info for info in module.imported_names
-                                     if info.name not in STDLIB_MODNAMES_SET]
             module.imports -= STDLIB_MODNAMES_SET
```

Comment:
I think this is probably no longer necessary, given the filtering done to imported_names before computing module.imports.

OTOH it does no harm to leave this in.